File: tasks/task2/monolith_client.py
import os
import httpx

MONOLITH_BASE = os.getenv("MONOLITH_BASE", "http://monolith:8084/api")

async def get_user(user_id: str):
    url = f"{MONOLITH_BASE}/users/{user_id}"
    async with httpx.AsyncClient() as client:
        r = await client.get(url)
        r.raise_for_status()
        return r.json()

async def get_hotel(hotel_id: str):
    url = f"{MONOLITH_BASE}/hotels/{hotel_id}"
    async with httpx.AsyncClient() as client:
        r = await client.get(url)
        r.raise_for_status()
        return r.json()

async def get_promo(promo_code: str):
    if not promo_code:
        return {"discountPercent": 0}
    url = f"{MONOLITH_BASE}/promos/{promo_code}"
    async with httpx.AsyncClient() as client:
        r = await client.get(url)
        if r.status_code == 404:
            return {"discountPercent": 0}
        r.raise_for_status()
        return r.json()

File: tasks/task2/booking-service/src/main/java/com/hotelio/booking/MonolithClient.java
package com.hotelio.booking;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;

import java.util.Map;

@Component
public class MonolithClient {

    private final RestTemplate restTemplate = new RestTemplate();

    @Value("${app.monolith.base-url}")
    private String baseUrl;

    public Map getUser(String userId) {
        return restTemplate.getForObject(baseUrl + "/api/users/" + userId, Map.class);
    }

    public Map getHotel(String hotelId) {
        return restTemplate.getForObject(baseUrl + "/api/hotels/" + hotelId, Map.class);
    }

    public Boolean isTrusted(String hotelId) {
        ResponseEntity<Boolean> response =
                restTemplate.getForEntity(baseUrl + "/api/reviews/hotel/" + hotelId + "/trusted", Boolean.class);
        return response.getBody();
    }

    public Map validatePromo(String code, String userId) {
        return restTemplate.postForObject(
                baseUrl + "/api/promos/validate?code=" + code + "&userId=" + userId,
                null,
                Map.class
        );
    }
}